src/window.rs

```diff
@@ -60,46 +59,45 @@ impl Window {
         };
 
         if self.frame_count == 0 {
+            // NOTE(mbernat): It's possible to avoid initial mode setting
+            // since we're keeping the previous mode: we could just call page_flip directly.
             self.drm_display.set_mode_with_framebuffer(Some(fb));
         } else {
             self.drm_display.page_flip(fb);
         }
+
+        // NOTE(mbernat): This buffer object returns to the surface's queue upon destruction.
```

Comment:
Where is this behavior documented? I didn't see a `Drop` impl for `gbm::BufferObject`. I believe you on the behavior, I'm just trying to follow the underlying code and put the pieces together.

Reply:
Unfortunately, I haven't found any documentation for this.

As for where it is implemented: https://docs.rs/gbm/latest/src/gbm/surface.rs.html#89
I haven't tracked that logic in detail but it's registering a callback with `gbm_surface_release_buffer`, which is presumably triggered by the buffer object's destruction. The C implementations call this function manually.